.clang-tidy

```diff
@@ -1,42 +1,32 @@
 Checks: "-*,\
 bugprone-*,\
+cert-dcl21-cpp,\
+cert-dcl50-cpp,\
+cert-env33-c,\
+-cert-err34-c,\
+cert-err52-cpp,\
+-cert-err58-cpp,\
+cert-err60-cpp,\
+-cert-flp30-c,\
 -clang-analyzer-security.FloatLoopCounter,\
 google-build-namespaces,\
--google-explicit-constructor,\
+google-explicit-constructor,\
 google-readability-todo,\
 google-runtime-operator,\
+misc-new-delete-overloads,\
+misc-non-copyable-objects,\
+misc-static-assert,\
 misc-suspicious-*,\
+misc-throw-by-value-catch-by-reference,\
 misc-unused-*,\
 modernize-loop-convert,\
 modernize-make-shared,\
 modernize-make-unique,\
 modernize-pass-by-value,\
 modernize-use-*,\
--performance-*,\
+performance-*,\
+readability-avoid-const-params-in-decls,\
 readability-braces-around-statements,\
 readability-container-size-empty,\
--readability-identifier-naming,\
```

Comment:
We should work out what to do here, at one point this check was enabled. Although it might be “dead” code at the moment I think it is useful to show intent

Reply:
My recommendation is that we enforce naming conventions using an in-house script. This goes against my every instinct, but this check is simply too poorly designed. I discussed this with @troelsfr. In brief: clang-tidy supports identifier affixes based on member access modifiers. We would like to use this and mark private and protected members with underscore. Here's the problem: clang-tidy only recognises non-static and non-const members for the purposes of this check. You can specify affixes for statics and const members, but they would be applied irrespective of access being public, private or protected.

```
class Foo
{
  int       private_member_with_suffix_;
  int const private_member_without_suffix; // inconsistent
};
```

I'll take care of this in the near future; Troels wrote a script that may serve as a starting point, though it's not quite ready to be tied into CI. For now I focused on strengthening checks that directly affect API behaviour, such as explicit constructors, performance checks, etc.